pkg/volume/iscsi/iscsi_util.go

```diff
@@ -212,6 +213,26 @@ func (util *ISCSIUtil) loadISCSI(conf *iscsiDisk, mnt string) error {
 	return nil
 }
 
+// scanLun scans a single LUN on one SCSI bus
+func scanLun(hostNumber int, lunNumber int) error {
+	filename := fmt.Sprintf("/sys/class/scsi_host/host%d/scan", hostNumber)
+	fd, err := os.OpenFile(filename, os.O_WRONLY, 0)
+	if err != nil {
+		return err
+	}
+	defer fd.Close()
+
+	scanCmd := fmt.Sprintf("0 0 %d", lunNumber)
```

Comment:
why `0 0 l`? if the channel and target are not 0, 0, this won't work. https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/5/html/online_storage_reconfiguration_guide/adding_storage-device-or-path

does this function help? https://github.com/NetApp/kubernetes/blob/04cac2212bdfa65ce717cf72242a189d96878352/pkg/volume/fc/fc_util.go#L123

Reply:
The FC scan function you linked is equivalent to the old behavior, and is what causes the bug. Writing 3 hypens "- - -" to the scan file causes the SCSI subsystem to scan the whole bus, finding disks that are not relevant to the current attach operation. The whole point of this new method is that is scans only a single LUN. I can expand the comment to make this more clear.